packages/shared/src/components/history/ReadingHistoryList.tsx

```diff
@@ -65,8 +77,17 @@ function ReadHistoryList({
     <section className="flex relative flex-col">
       {renderList()}
       <InfiniteScrollScreenOffset ref={infiniteScrollRef} />
+      <PostOptionsReadingHistoryMenu
+        post={readingHistoryContextItem?.post}
+        onHidden={() => setReadingHistoryContextItem(null)}
+        onHide={(postId) =>
+          onHide({
+            postId,
+            timestamp: readingHistoryContextItem.timestampDb,
+            ...queryIndexes,
+          })
```

Comment:
This seems a bit messy, can we not access al these things from the readingHistoryContextItem?

Reply:
Agree, it is a bit messy, but ReadingHistoryContextItem doesn't contain queryIndexes at the moment. Do you suggest to refactor the useReadingHistoryContextMenu hook to include queryIndexes inside the Item?

Reply:
Unfortunately, we can't get this through `readingHistoryContextItem` as the values needed are part of the data pages array. I think another way is to loop through every item to search for this specific history though I think it would be too much to loop through again. We can definitely refactor this once we can find another viable solution for this scenario but for now, imho this should be alright.